### PR TITLE
feat(analytics): track custom domain identifier

### PR DESCRIPTION
**Pull request by james-a-morris**
This PR allows custom tracking of domain ids when a user includes the `?integrator_id` query param.

### CODE REVIEW

File: src/hooks/useReferrer.ts [hunk 1]
```diff
@@ -46,8 +46,24 @@ export default function useReferrer() {
       setIsResolved(true);
     }
   }, [provider, r]);
+
+  // Return the integratorId as a hex string with a 0x prefix if not already present.
+  // Default to Across if no integrator_id is provided or if the integrator_id is invalid.
+  const integratorId = useMemo(() => {
```
Comment (gsteenkamp89) [inline src/hooks/useReferrer.ts:52]: Does this need to be memoized?

**Review by gsteenkamp89 — APPROVED** [review]

File: src/hooks/useReferrer.ts (hunk 1 above)
Comment (dohaki) [reply src/hooks/useReferrer.ts:52]: Yea, I don't think we need to

Comment (james-a-morris) [reply src/hooks/useReferrer.ts]: resolved. Will merge

**Comment by james-a-morris** [comment]
Merging without the memoization

**Pull request merged**